app/public/static/js/components/keypad.js

```diff
@@ -0,0 +1,49 @@
+AFRAME.registerComponent("keypad", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+  multiple: true,
+  /**
+   * Initial creation and setting of the mesh.
+   */
+  init: function () {
+    const data = this.data.jsonData;
+    const password = data.password;
+    const el = this.el;
+    el.setAttribute(
+      "super-keyboard",
+      "imagePath:/static/img/; multipleInputs:true; model: numpad; align: center; maxLength: 4; label: Enter Password; labelColor: black"
+    );
+    el.setAttribute("scale", { x: 20, y: 20, z: 20 });
+    el.setAttribute("position", { x: 0, y: 0, z: 0.5 });
+
+    el.setAttribute("class", "link");
+
+    el.addEventListener("superkeyboardinput", function (event) {
+      if (event.detail.value === password) {
+        el.setAttribute(
+          "super-keyboard",
+          "label:SUCCESS; labelColor: green; multipleInputs:true"
+        );
```

Comment:
Personally I think auto closing (after 1.5 seconds or so) on a success makes sense. But I'll leave the decision to you guys

Reply:
Synced offline about this; the success workflow will be worked on later once we have state